src/graph/Graph.cpp

```diff
@@ -201,6 +206,41 @@ void Graph::dijkstraShortestPath(Vertex *s, vector<Vertex *> dests) {
     }
 }
 
+/**
+ * Similar to dijkstraShortestPath, but stops when it finds the first client (closest)
+ * @param s
+ * @param dests
+ */
+Client* Graph::dijkstraClosestClient(Vertex *s, vector<Vertex *> dests) {
+    for (Vertex* v : vertexSet) {
+        v->dist = INF;
+        v->path = nullptr;
+    }
+    s->dist = 0;
+    MutablePriorityQueue<Vertex> q;
+    q.insert(s);
+
+    while (!q.empty()) {
+        Vertex* v = q.extractMin();
+
+        // Check if we reached any vertices
+        vector<Vertex*>::iterator it;
+        if ((it = find(dests.begin(), dests.end(), v)) != dests.end()) {
+            return v->client;   // Found a Client
+        }
+
+        for (Edge e : v->adj) {
+            double oldDist = e.dest->dist;
+            if (relax(v, e)) {
+                if (oldDist == INF) q.insert(e.dest);
+                else q.decreaseKey(e.dest);
+            }
+        }
+    }
+
+    return NULL;
+}
+
```

Comment:
I understand the idea but I don't think this works well. The point of the nearest neighbour is to find the client with the lowest cost. With this implementation, we're just getting the client that the Dijkstra's algorithm happens to find first (like a BFS), we don't have any guarantee that that client has the lowest cost right now, meaning that we'll probably have a much worse solution in the end